### PR TITLE
Security team requirements

### PR DESCRIPTION
**Pull request by misohu**
Add security team requirements to the image:
- Create security manifest
- Refactor the repositories installation with [these](https://docs.google.com/document/d/1piO09xRbvwEwfru-oOgBpctA47x8kX_S2Fr7Y2ovHfs/edit#) recommendations

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,11 +1,24 @@
-FROM ubuntu:22.04
+FROM ubuntu:jammy
```
Comment (beliaev-maksim) [inline Dockerfile:1]: why jammy instead of version ?

File: Dockerfile [hunk 2]
```diff
@@ -1,11 +1,24 @@
...
+    ; \
+    DEBIAN_FRONTEND=noninteractive apt-get remove --purge --auto-remove -y; \
+    rm -rf /var/lib/apt/lists/*; \
+# install mlflow
+    pip install --no-cache mlflow==$VERSION; \
+# create manifest for security scan
+    mkdir -p /usr/share/rocks; \
+    (echo "# os-release" && cat /etc/os-release && echo "# dpkg-query" && dpkg-query -f '${db:Status-Abbrev},${binary:Package},${Version},${source:Package},${Source:Version}\n' -W) > /usr/share/rocks/dpkg.query
 
+EXPOSE 5000
```
Comment (beliaev-maksim) [inline Dockerfile:23]: is it mlflow port?

File: Dockerfile [hunk 3]
```diff
@@ -1,11 +1,24 @@
...
-    apt clean
+    python3-pip \
+    tzdata \
+    ; \
+    DEBIAN_FRONTEND=noninteractive apt-get remove --purge --auto-remove -y; \
+    rm -rf /var/lib/apt/lists/*; \
+# install mlflow
+    pip install --no-cache mlflow==$VERSION; \
+# create manifest for security scan
+    mkdir -p /usr/share/rocks; \
```
Comment (beliaev-maksim) [inline Dockerfile:20]:
```suggestion
    rm -rf /var/lib/apt/lists/*; \
    # install mlflow
    pip install --no-cache mlflow==$VERSION; \
    # create manifest for security scan
    mkdir -p /usr/share/rocks; \
```

File: Dockerfile [hunk 4]
```diff
@@ -1,11 +1,24 @@
...
 ARG VERSION
 
-RUN apt update && apt install -y \
+ENV TZ=UTC
+
+RUN set -eux; \
+# install python
+    apt-get update; \
+    DEBIAN_FRONTEND=noninteractive apt-get full-upgrade -y; \
+    DEBIAN_FRONTEND=noninteractive apt-get install -y --no-install-recommends \
```
Comment (beliaev-maksim) [inline Dockerfile:10]: don't we miss `;` or `&&` at the end ?

File: Dockerfile [hunk 5]
```diff
@@ -1,11 +1,24 @@
...
     python3.10 \
-    python3-pip && \
-    pip install --no-cache mlflow==$VERSION && \
-    apt autoremove -yqq --purge && \
-    apt clean
+    python3-pip \
+    tzdata \
+    ; \
```
Comment (beliaev-maksim) [inline Dockerfile:14]:
```suggestion
    tzdata; \
```

File: Dockerfile (hunk 1 above)
Comment (misohu) [reply Dockerfile:1]: its in the recommentations https://docs.google.com/document/d/1piO09xRbvwEwfru-oOgBpctA47x8kX_S2Fr7Y2ovHfs/edit#heading=h.g20qrwjts0x

File: Dockerfile (hunk 4 above)
Comment (misohu) [reply Dockerfile:10]: recommentation says to use ; instead of &&

Comment (beliaev-maksim) [reply Dockerfile:10]: ok, you do not have any at the end of the line

File: Dockerfile (hunk 3 above)
Comment (beliaev-maksim) [reply Dockerfile:20]: for me that looks weird, if you write python, you probably will never do like this

File: Dockerfile (hunk 4 above)
Comment (misohu) [reply Dockerfile:10]: true lol

File: Dockerfile (hunk 3 above)
Comment (misohu) [reply Dockerfile:19]: I dont know I think to stay consistent I will leave it as this

File: Dockerfile (hunk 4 above)
Comment (misohu) [reply Dockerfile:10]: lol no ... it is installing packages on next lines

Comment (beliaev-maksim) [reply Dockerfile:10]: I see now, we do double export of the variable, that one tricked me

File: Dockerfile (hunk 3 above)
Comment (cjdcordeiro) [reply Dockerfile:19]: I don't think you should focused too much on how others are doing things :) cause maybe you'll have the chance to do them better!

Anyway, I'd also indent the comments, but that's a nitpick. as you prefer

Comment (cjdcordeiro) [reply Dockerfile:19]: btw, one suggestion I would make as well, is to move this metadata file creation into a separate layer. Why? Well, you seemed a bit concerned about diverging this Dockerfile from the upstream one. If you have 2 layers you'll at least keep the 1st one aligned with upstream. Just a design choice...should harm the end result. It's up to you

File: Dockerfile [hunk 6]
```diff
@@ -1,11 +1,23 @@
 FROM ubuntu:22.04
```
Comment (cjdcordeiro) [inline Dockerfile:1]: another couple of suggestions:
 - ECR is updated more frequently than the official Ubuntu image in Docker Hub, so if you want to use that one instead, it's `public.ecr.aws/ubuntu/ubuntu:22.04`
 - if you're planning on using `renovate` in this repo, then you might want to ping a digest, i.e. `ubuntu:22.04@${digest}`

**Review by cjdcordeiro — APPROVED** [review]
It lgtm. Left a few suggestions but they are just that, suggestions. 

Nice work

File: Dockerfile [hunk 7]
```diff
@@ -1,11 +1,22 @@
...
-    pip install --no-cache mlflow==$VERSION && \
-    apt autoremove -yqq --purge && \
-    apt clean
+    python3-pip \
+    tzdata; \
+    DEBIAN_FRONTEND=noninteractive apt-get remove --purge --auto-remove -y; \
+    rm -rf /var/lib/apt/lists/*; \
+    # install mlflow
+    pip install --no-cache mlflow==$VERSION
+RUN mkdir -p /usr/share/rocks; \
```
Comment (beliaev-maksim) [inline Dockerfile:18]: please add a comment in the file explaining why we do it in the separate layer

Comment (beliaev-maksim) [reply Dockerfile:18]: once done, approved

**Review by beliaev-maksim — APPROVED** [review]

**Pull request merged**